cmd/compute-domain-kubelet-plugin/driver.go

```diff
@@ -171,73 +173,75 @@ func (d *driver) NodeUnprepareResources(ctx context.Context, req *drapbv1.NodeUn
 	d.Lock()
 	defer d.Unlock()
 
-	resourceClaim, err := d.client.ResourceV1beta1().ResourceClaims(claim.Namespace).Get(
-		ctx,
-		claim.Name,
-		metav1.GetOptions{})
-	if err != nil {
-		ret := &drapbv1.NodePrepareResourceResponse{
-			Error: fmt.Sprintf("failed to fetch ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
-		}
-		return isPermanentError(err), ret
-	}
-
-	if resourceClaim.Status.Allocation == nil {
-		ret := &drapbv1.NodePrepareResourceResponse{
-			Error: fmt.Sprintf("no allocation set in ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
+	if claim.Status.Allocation == nil {
+		res := kubeletplugin.PrepareResult{
+			Err: fmt.Errorf("no allocation set in ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
 		}
-		return true, ret
+		return true, res
 	}
 
-	prepared, err := d.state.Prepare(ctx, resourceClaim)
+	devs, err := d.state.Prepare(ctx, claim)
 	if err != nil {
-		ret := &drapbv1.NodePrepareResourceResponse{
-			Error: fmt.Sprintf("error preparing devices for claim %v: %v", claim.UID, err),
+		res := kubeletplugin.PrepareResult{
+			Err: fmt.Errorf("error preparing devices for claim %v: %v", claim.UID, err),
 		}
-		return isPermanentError(err), ret
+		return isPermanentError(err), res
+	}
+
+	// Translate type: drapbv1.Device -> kubeletplugin.Device
+	// Maybe instead change return type of state.Prepare()
+	var prepDevs []kubeletplugin.Device
+	for _, d := range devs {
+		device := kubeletplugin.Device{
+			// kubeletplugin/draplugin.go has a stripSubrequestNames() helper:
+			// Requests: stripSubrequestNames(d.RequestNames),
+			Requests:     d.RequestNames,
+			PoolName:     d.PoolName,
+			DeviceName:   d.DeviceName,
+			CDIDeviceIDs: d.CDIDeviceIDs,
+		}
+		prepDevs = append(prepDevs, device)
 	}
 
-	klog.Infof("Returning newly prepared devices for claim '%v': %v", claim.UID, prepared)
-	return true, &drapbv1.NodePrepareResourceResponse{Devices: prepared}
+	klog.Infof("Returning newly prepared devices for claim '%v': %v", claim.UID, prepDevs)
+	return true, kubeletplugin.PrepareResult{Devices: prepDevs}
 }
 
-func (d *driver) nodeUnprepareResource(ctx context.Context, claim *drapbv1.Claim) (bool, *drapbv1.NodeUnprepareResourceResponse) {
+func (d *driver) nodeUnprepareResource(ctx context.Context, claimNs kubeletplugin.NamespacedObject) (bool, error) {
 	d.Lock()
 	defer d.Unlock()
 
-	resourceClaim, err := d.client.ResourceV1beta1().ResourceClaims(claim.Namespace).Get(
+	claim, err := d.client.ResourceV1beta1().ResourceClaims(claimNs.Namespace).Get(
 		ctx,
-		claim.Name,
+		claimNs.Name,
 		metav1.GetOptions{})
+
 	if err != nil {
-		ret := &drapbv1.NodeUnprepareResourceResponse{
-			Error: fmt.Sprintf("failed to fetch ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
-		}
-		return isPermanentError(err), ret
+		return isPermanentError(err), fmt.Errorf(
+			"failed to fetch ResourceClaim %s in namespace %s: %v",
```

Comment:
```suggestion
			"failed to fetch ResourceClaim %s in namespace %s: %w",
```
it is good practice to wrap errors so as to allow `error.Is` check at callsites when needed.

Reply:
Unless the underlying error is an implementation detail that the caller should not be exposed to...

In general this nuance is typically ignored in favor of the more easily reviewed and enforced "always wrap".

Reply:
Will do.

I had a commit on this branch where I changed all `%s` and `%v` to `%w` in `fmt.Errorf()` err formatting calls -- then I backed out of it to keep the diff smaller.

Reply:
done